processor/src/stack/mod.rs

```diff
@@ -68,11 +68,34 @@ impl Stack {
         init_trace_capacity: usize,
         keep_overflow_trace: bool,
     ) -> Self {
+        let init_values = inputs.stack_init();
+        let depth = cmp::max(MIN_STACK_DEPTH, init_values.len());
+
+        let (trace, overflow) = if init_values.len() > MIN_STACK_DEPTH {
+            let overflow = OverflowTable::new_with_inputs(
+                keep_overflow_trace,
+                &init_values[MIN_STACK_DEPTH..],
+            );
+            let trace = StackTrace::new(
+                &init_values[..MIN_STACK_DEPTH],
+                init_trace_capacity,
+                depth,
+                Felt::new(Felt::MODULUS - 1),
```

Comment:
nit: I would have probably used `-ONE` instead of `Felt::new(Felt::MODULUS - 1)`. But we can change this later.